### PR TITLE
Add contribution guidelines that github will automatically display

### PR DESCRIPTION
**Pull request by mwcraig**
I thought this might be helpful in ensuring that contributions follow the workflow you recommend and your commit message convention. Github will automatically display a link to a file called `CONTRIBUTING` or `CONTRIBUTING.md` if it exists in the base of the repo. 

More information about this github feature is at: https://help.github.com/articles/setting-guidelines-for-repository-contributors/

Let me know if you want any revisions to this!

### CODE REVIEW

**Comment by rgommers** [comment]
Thanks @mwcraig, good idea to add this file.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Contributing to numpy
+
+## Reporting issues
+
+When reporting issues please include as much detail as possible about your
+operating system, numpy version and python version. Whenever possible, please
+also include a brief code example that demonstrates the problem.
```
Comment (rgommers) [inline CONTRIBUTING.md]: maybe add "self-contained" after brief.

Comment (rgommers) [inline CONTRIBUTING.md]: And add that for segfaults a Gdb traceback should be added, plus a link on how to do that (https://github.com/numpy/numpy/blob/master/doc/source/dev/development_environment.rst#debugging)

**Comment by charris** [comment]
Thanks @mwcraig .

**Pull request merged**